text/0041-Recoverable-fee-history-for-FOG-compatible-gift-codes.md

```diff
@@ -0,0 +1,137 @@
+- Feature Name: Recoverable-fee-history-For-FOG-compatible-gift-codes
+- Start Date: May 18
+- MCIP PR: [mobilecoinfoundation/mcips#0041](https://github.com/mobilecoinfoundation/mcips/pull/0041)
+- Tracking Issue: https://github.com/mobilecoinfoundation/mobilecoin/issues/2001
+
+# Summary
+[summary]: #summary
+
+[MCIP #32 - FOG-compatible-gift-codes](https://github.com/mobilecoinfoundation/mcips/blob/main/text/0032-fog-compatible-gift-codes.md) 
+outlines a proposal for TxOut based gift codes which FOG users can privately create, read, and redeem without having to 
+possess a copy of the ledger or rely on an untrusted FOG ledger endpoint. This enables gift codes to be used by mobile 
+wallet users and potentially be sent to individuals who do not yet have a MobileCoin account.
+
+This is achieved in 2 stages:
+1. Sending a TxOut to a special subaddress reserved for gift codes at u64::MAX - 2 along with a *funding memo* written 
+to the change output of the transaction that records the first 4 bytes of a Blake2b512 hash of the gift code TxOut and a
+60 byte utf-8 note that can be used to describe information about the gift code. If the sending party decides to cancel 
+this gift code TxOut (say, if it is never redeemed by a receiver) it is sent back to the sending party's change address with 
+a *cancellation memo* that includes the global index of the gift code TxOut originally sent to the reserved gift code 
+subaddress.
+
+
+2. Sending a protobuf message via grpc to another party giving the information required to locate, unmask, and spend the 
+TxOut. The receiving party "redeems" the gift code by sending it to their change subaddress along with a *sender memo* 
+that allows the receiver to record up to 64 utf-8 bytes to document any desired information about the gift code.
+
+The memos which record this information are shown in the table below:
+
+| Memo type bytes | Name                                              |
+| -----------     | -----------                                       |
+| 0x0201          | Gift code funding memo                            |
+| 0x0202          | Gift code cancellation memo                       |
+| 0x0002          | Gift code sender memo                       |
+
```

Comment:
I don't think there's a need to repeat all this information here, it's available in MCIP 32.

Reply:
Slimmed down, but I think it's worth keeping the table.